### DISCUSSION TITLE
Creating a sector-specific Information Model by using IDS Information Model

### DISCUSSION THREAD

**Discussion by anilturkmayali** [IndustrialDataSpace/InformationModel#589] [Q&A]
Hi all, 

I would like to ask what would be the best steps to follow if someone wants to create a sector-specific information model, based on IDS Information Model? 

There is a EU project, where partners will be looking to implement an information model for agrifood industry, and I was wondering how we can direct them to the easiest source of information? Such as a step-by-step guide, if possible.

Thank you

**Comment by clange** [comment]
Hi Anil,

this question comes just at the right time that we have finished the documentation of how to develop application profiles based on the IDS Information Model. Consider "Application Profiles" a synonym for "sector-specific information models".

* Our results are documented in this repository: https://github.com/International-Data-Spaces-Association/IM-Application-Profiles (contributed by @lcomet @JohannesLipp)
* Here's the technical core: SHACL shapes that implement all validation rules in a machine-actionable way: https://github.com/International-Data-Spaces-Association/IM-Application-Profiles/tree/main/base. To validate a given instance, you combine all relevant shapes (or you simply take _all_ of them) and use an off-the-shelf SHACL validator. E.g., for mandatory properties (metadata fields etc.), pay attention to `sh:minCount 1`. E.g., for a concrete example, see https://github.com/International-Data-Spaces-Association/IM-Application-Profiles/blob/main/base/communication/EndpointShape.ttl
* In that repository (rather than the core infomodel repository) we have further examples – here is one actually about domain-specific extensions: https://github.com/International-Data-Spaces-Association/IM-Application-Profiles/tree/main/examples-SHACL-validations 
* Finally, here is a validator demo developed by @fabiankirstein: https://ids-ap.fokus.fraunhofer.de/ – When you click "Editor" you can load the "basic" shapes (plain IDS) or the "extended" shapes (including domain-specific adaptations) to validate against. The idea is that both examples remain compatible with the core IDS information model, while the "extended" version adds application-specific functionality.

**Reply by anilturkmayali** [reply]
Hi Christoph, 
Thank you very much for the great and detailed explanation. The sources you have shared will be highly beneficial for some projects (also from other domains). I will share them with the project teams, hopefully they will adopt IDS InfoModel and may also provide feedback based on their experimentation.

**Comment by rob-metalinkage** [comment]
Hi,

great to see this important topic progressing (finally!). 

Am involved in a number of projects where integration with IDS based systems is intended, and the OGC I represent is interested in publishing spatio-temporal profiles of IDS to support these projects. If the community is interested then these can be formally introduced into the OGC standards program.

 In the W3C Data Exchange Working Group (which defines DCAT) the matter of profiling DCAT has been raised, but mechanisms and best practices for this have not been satisfactorily progressed - a draft best practice was attempted but insufficient experience and understanding of the functional nature of profiles was available for the group to commit to any specific direction. An early decision _not_ to publish any specific profiles of DCAT but focus on the core tactically avoided the need to address this in the short term, however as this thread evidences there is clearly a need for a best practice.

Nevertheless, analysis of the Use Cases and Requirements (https://w3c.github.io/dxwg/ucr/) for the DCAT revision highlighted the need for formal description of the sort of profiles hierarchy implied by profiling IDS,  which in turn profiles (some aspects only) of DCAT.   The response to this was a draft ontology for describing profiles, their relationships and the roles and form of the sort of resources.  Please look at https://www.w3.org/TR/dx-prof/  to find the latest W3C Note for the Profiles Vocabulary (PROF).

The motivation for engaging with DXWG was for the Open Geospatial Consortium to support communities to publish profiles of OGC standards, but also allow for OGC governance of common spatio-temporal profiles of more general standards.  Apropos of this, we are embarking on the publication of GeoDCAT, as a general purpose profile of DCAT. This will support the more specific and pre-existing GeoDCAT-AP profile and be available for other potential uses outside EU portals scope of DCAT-AP. 

We will be using PROF to publish a register of profiles - we have a version in development here: 

http://defs-dev.opengis.net/profiles/vocab/  (please refresh if you get an error - we have a update to the UI planned and havent got round to fixing this)

as an example, consider a minimal profile of SKOS suitable for sharing terms - 
http://defs-dev.opengis.net/profiles/object?uri=http%3A//www.opengis.net/def/metamodel/profiles/skos_shared
(when deployed this will resolve directly from http://www.opengis.net/def/metamodel/profiles/skos_shared)

this has various flavours of SHACL and validation examples exposed using the PROF vocabulary.


In terms of a specific forum to progress this, we would like to restart the PROF vocabulary sub-group of DXWG if we have active community members ready to consider the needs for formalised profiles.
